arrows-stdlib-cats/src/main/scala/arrows/stdlib/cats.scala

```diff
@@ -0,0 +1,114 @@
+package arrows.stdlib
+
+import cats.{ Applicative, StackSafeMonad }
+import cats.effect._
+import cats.arrow.ArrowChoice
+import cats.kernel.{ Monoid, Semigroup }
+import cats.mtl.ApplicativeAsk
+
+import scala.concurrent.{ ExecutionContext, Promise }
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success }
+
+object Cats extends ArrowInstances
+
+sealed abstract class ArrowInstances extends ArrowInstances1 {
+
+  implicit def catsEffectForTask(implicit ec: ExecutionContext): Effect[Task] = new ArrowAsync[Unit] with Effect[Task] {
+    def runAsync[A](fa: Task[A])(cb: Either[Throwable, A] => IO[Unit]): SyncIO[Unit] =
+      SyncIO { fa.run(())(ec); () }
+  }
+
+  implicit val catsArrowChoiceForArrow: ArrowChoice[Arrow] = new ArrowChoice[Arrow] {
+    def choose[A, B, C, D](f: Arrow[A, C])(g: Arrow[B, D]): Arrow[Either[A, B], Either[C, D]] =
+      Arrow[Either[A, B]].flatMap {
+        case Left(a)  => f(a).map(Left(_))
+        case Right(b) => g(b).map(Right(_))
+      }
+
+    def lift[A, B](f: A => B): Arrow[A, B] = Arrow[A].map(f)
+
+    def first[A, B, C](fa: Arrow[A, B]): Arrow[(A, C), (B, C)] =
+      Arrow[(A, C)].flatMap { case (a, c) => fa(a).map(_ -> c) }
+
+    def compose[A, B, C](f: Arrow[B, C], g: Arrow[A, B]): Arrow[A, C] = g.andThen(f)
+  }
+
+  implicit def catsApplicativeAskForArrow[E]: ApplicativeAsk[Arrow[E, ?], E] = new ApplicativeAsk[Arrow[E, ?], E] {
```

Comment:
could it be a `val`?

Reply:
It can't be a `val` because then you can't define the `E` type parameter :)

Reply:
we could cache it and cast:

```scala
private final val catsApplicativeAskForArrowInstance: ApplicativeAsk[Arrow[Any, ?], Any] ...
implicit def catsApplicativeAskForArrow[E] = catsApplicativeAskForArrowInstance.asInstanceOf[ApplicativeAsk[Arrow[E, ?], E]]
```